pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -253,6 +256,19 @@ const FAILED_DOWNLOAD_RETRIES: u32 = 10;
 // retries. Uploads and deletions are retried forever, though.
 const FAILED_UPLOAD_WARN_THRESHOLD: u32 = 3;
 
+pub enum MaybeDeletedIndexPart {
+    IndexPart(IndexPart),
+    Deleted,
+}
+
+/// Errors that can arise when calling [`RemoteTimelineClient::stop`].
+#[derive(Debug, thiserror::Error)]
+pub enum StopError {
+    /// Callers are responsible for checking this before calling `stop()`.
+    #[error("queue is not initialized")]
+    QueueUninitialized,
+}
```

Comment:
Caller is not checking against this in http handler -> `Tenant::delete_timeline`, should it be?

Also this operation needs to be idempotent.